packages/embark/src/lib/modules/blockchain_connector/index.js

```diff
@@ -598,8 +600,22 @@ class BlockchainConnector {
               // FIXME Returns an error because we are too low
               return eachCb();
             }
-            blocks.push(block);
-            eachCb();
+            if (returnTransactionObjects && includeTransactionReceipts) {
+              Promise.all(block.transactions.map(tx => (
+                self.web3.eth.getTransactionReceipt(tx.hash)
+              )))
+                .then(receipts => {
+                  block.transactions.forEach((tx, index) => {
+                    tx['receipt'] = receipts[index];
+                  });
+                  blocks.push(block);
+                  eachCb();
+                })
+                .catch(() => eachCb());
+            } else {
+              blocks.push(block);
+              eachCb();
```

Comment:
Could you reverse the condition and early return?
```
if (!returnTransactionObjects || !includeTransactionReceipts) {
  blocks.push(block);
  return eachCb();
}
```

Reply:
Done.